### PR TITLE
Ensure resource is resolved when picking children

### PR DESCRIPTION
**Pull request by alexweininger**
Needed to completely fix https://github.com/microsoft/vscode-azurefunctions/issues/3135

### CODE REVIEW

File: src/tree/ResolvableTreeItemBase.ts [hunk 1]
```diff
@@ -63,6 +63,13 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
         });
     }
 
+    public async pickTreeItemImpl(expectedContextValues: (string | RegExp)[], context: IActionContext): Promise<AzExtTreeItem | undefined> {
+        if (!this.resolveResult) {
+            await this.resolve(true, context);
+        }
+        return super.pickTreeItemImpl?.(expectedContextValues, context);
+    }
```
Comment (alexweininger) [inline src/tree/ResolvableTreeItemBase.ts:71]: Should we be resolving these in a `pickTreeItemImpl` in `GroupTreeItemBase`?

Comment (nturinski) [reply src/tree/ResolvableTreeItemBase.ts:71]: Yeah, I think that makes more sense.

We can either have a loading bar while the children resolve or just display the list of current children and have them resolve in the background, and then await it if it hasn't finished resolving by the time the user selects it.

**Review by nturinski — APPROVED** [review]

**Comment by alexweininger** [comment]
Unsure if this is still needed since we went a different direction #212, will leave it up for now.

**Comment by nturinski** [comment]
It shouldn't be since the app resource is being resolved after it's picked now in #212

**Pull request closed without merging**